spec/helpers/submission_helper_spec.rb

```diff
@@ -37,4 +37,30 @@
       end
     end
   end
+
+  describe '#get_file_info' do
+    let(:assignment) { create(:assignment) }
+    let(:grouping) { create(:grouping, assignment: assignment) }
+    let(:file_name) { 'test.zip' }
+    before do
+      grouping.group.access_repo do |repo|
+        txn = repo.get_transaction('test')
+        path = File.join(grouping.assignment.repository_folder, file_name)
+        txn.add(path, '')
+        repo.commit(txn)
+      end
+    end
+    it 'should generate and return the file url with correct assignment id' do
+      file_info = {}
+      grouping.group.access_repo do |repo|
+        revision = repo.get_latest_revision
+        full_path = File.join(grouping.assignment.repository_folder, '')
+        file_obj = revision.tree_at_path(full_path)[file_name]
+        dirname, basename = File.split(file_name)
+        dirname = '' if dirname == '.'
+        file_info = get_file_info(basename, file_obj, assignment.id, revision.revision_identifier, dirname, grouping.id)
+      end
+      expect(file_info[:url].include?(assignment.id.to_s)).to be true
```

Comment:
Parse the url and make sure the assignment_id is actually correct. `include?` over the whole url string is too general.